Main/ClientApp/src/Components/Dialogs/EditContributorDialog.tsx

```diff
@@ -0,0 +1,169 @@
+import SimpleDialog from "./SimpleDialog";
+import { updateSharedCollection } from "../../Actions/CollectionActions";
+import {
+  Button,
+  Avatar,
+  MenuItem,
+  FormControl,
+  InputLabel,
+  Select,
+  makeStyles,
+  Theme,
+  createStyles,
+  Grid,
+} from "@material-ui/core";
+import { Person } from "@material-ui/icons";
+import {
+  SharedCollectionData,
+  SharedCollection,
+} from "../../Model/SharedCollection";
+import SharedCollectionStore from "../../Stores/SharedCollectionsStore";
+import DeleteContributorDialog from "./DeleteContributorDialog";
+
+type EditContributorDialogProps = {
+  open: boolean;
+  toggleDialogOpen: () => void;
+  collectionId: number;
+  userId: number;
+};
+
+enum UserRights {
+  ViewRights,
+  EditRights,
+}
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      marginBlockEnd: "10px",
+    },
+  })
+);
+
+export default function EditContributorDialog(
+  props: EditContributorDialogProps
+) {
+  const classes = useStyles();
+  const title = "Edit contributor";
+  const description = `You can change contributor permissions or delete this contributor.`;
+  const [
+    sharedCollection,
+    setSharedCollection,
+  ] = useState<SharedCollection | null>(null);
+  const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
+  const [selectedUserRights, setUserRights] = useState(
+    sharedCollection?.editRights ? UserRights.EditRights : UserRights.ViewRights
+  );
+
+  const createSharedCollectionData = () => {
+    return {
+      collectionId: props.collectionId,
+      userId: sharedCollection?.user.id,
+      editRights: selectedUserRights === UserRights.EditRights,
+    } as SharedCollectionData;
+  };
+
+  const onUserRightsChange = (event: React.ChangeEvent<{ value: any }>) => {
+    setUserRights(event.target.value);
+  };
+
+  useEffect(() => {
+    setSharedCollection(
+      SharedCollectionStore.getContributorSharedCollection(
+        props.collectionId,
+        props.userId
+      )
+    );
+    const changeHandler = () => {
+      setSharedCollection(
+        SharedCollectionStore.getContributorSharedCollection(
+          props.collectionId,
+          props.userId
+        )
+      );
+    };
+
+    SharedCollectionStore.addChangeListener(changeHandler);
+    return () => {
+      SharedCollectionStore.removeChangeListener(changeHandler);
+    };
+  }, [props.collectionId, props.userId, sharedCollection?.editRights]);
+
+  return (
+    <>
+      <SimpleDialog
+        open={props.open}
+        toggleDialogOpen={props.toggleDialogOpen}
+        title={title}
+        description={description}
+        content={
+          <Grid
+            container
+            spacing={3}
+            alignItems="center"
+            className={classes.root}
+          >
+            <Grid item>
+              <Avatar>
+                <Person />
+              </Avatar>
+            </Grid>
+            <Grid item xs={7}>
```

Comment:
I thought the xs chould add up to 12

Reply:
and if it is only defined on a single item, does it even change anything?

Reply:
it changes the width of that element, works without adding up to 12 so i left it